### ISSUE TITLE
Typecheck ReadBuiltFile Error

### ISSUE THREAD

**Issue by ivila** [golangci/golangci-lint#358]
mistake in recognize the built file path, golangci-lint think it should be "/tmp/gopackages113960511/go-build/net/cgo_linux.cgo1.go:1", but actually it is "/tmp/gopackages113960511/go-build/net/_usr_lib_golang_src_net_cgo_linux.cgo1.go"

Please include the following information:
1. Version of golangci-lint: 

87aae7794301cdffc1a8050a98acec9454a52dee

2. Config file: 

``` yml
run:
  skip-dirs:
    - vendor
    - bases/cat
    - bases/memcache
    - qconf

linters:
  disable:
    - megacheck
  enable:
    - golint
```

3. Go environment:
```
go version go1.9.5 linux/amd64
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/claude/.go"
GORACE=""
GOROOT="/usr/lib/golang"
GOTOOLDIR="/usr/lib/golang/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build257551088=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

4. Verbose output of running: 
```
INFO [config_reader] Config search paths: [./ /home/claude/.go/src/openapi /home/claude/.go/src /home/claude/.go /home/claude /home /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 8 linters: [deadcode errcheck golint govet ineffassign structcheck typecheck varcheck] 
INFO [loader] Go packages loading at mode load types and syntax took 3.163166983s 
INFO [loader/astcache] Parsed AST of all pkg.GoFiles: [/home/claude/.go/src/openapi/qconf/cache.go /home/claude/.go/src/openapi/qconf/tool.go /home/claude/.go/src/openapi/qconf/go_qconf.go] for 1.172504ms 
INFO [loader] Packages that do not compile: [openapi openapi/bases/cache openapi/bases/cat openapi/bases/cat/message [openapi/bases/cat/message.test] openapi/bases/cat/util openapi/bases/controller openapi/bases/memcache openapi/bases/redis openapi/controllers openapi/qconf openapi/routers openapi/utils] 
INFO [runner] worker.2 took 293.943533ms with stages: ineffassign: 149.174688ms, typecheck: 139.066505ms, govet: 2.600586ms, varcheck: 2.195895ms, deadcode: 577.317µs, structcheck: 138.692µs, errcheck: 128.419µs 
main.go:1: /tmp/gopackages113960511/go-build/net/cgo_linux.cgo1.go:1: no such file or directory (typecheck)
package main
bases/cache/cache.go:1: /tmp/gopackages113960511/go-build/net/cgo_linux.cgo1.go:1: no such file or directory (typecheck)
// Package cache 封装方法返回值缓存类
bases/controller/base.go:1: /tmp/gopackages113960511/go-build/net/cgo_linux.cgo1.go:1: no such file or directory (typecheck)
package controller
bases/redis/model.go:1: /tmp/gopackages113960511/go-build/openapi_vendor_zhangyue.com_architecture_zynsc_go_qconf/go_qconf.cgo1.go:1: no such file or directory (typecheck)
package redis
utils/common.go:1: /tmp/gopackages113960511/go-build/openapi_vendor_zhangyue.com_architecture_zynsc_go_qconf/go_qconf.cgo1.go:1: no such file or directory (typecheck)
package utils
INFO [runner] worker.1 took 606.292517ms with stages: golint: 606.259659ms 
INFO [runner] Workers idle times: #2: 312.260307ms 
INFO [runner/skip dirs] Skipped by pattern bases/cat dirs: [bases/cat/util bases/cat bases/cat/message bases/cat/util bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util bases/cat/util] 
INFO [runner/skip dirs] Skipped by pattern bases/memcache dirs: [bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache bases/memcache] 
INFO [runner/skip dirs] Skipped by pattern qconf dirs: [qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf qconf] 
INFO [runner/max_same_issues] 2/5 issues with text "/tmp/gopackages113960511/go-build/net/cgo_linux.cgo1.go:1: no such file or directory" were hidden, use --max-same-issues 
INFO [runner] Issues before processing: 276, after processing: 5 
INFO [runner] processing took 47.201547ms with stages: skip_dirs: 36.249833ms, exclude: 5.72279ms, path_prettifier: 2.477201ms, cgo: 1.361957ms, source_code: 786.595µs, nolint: 419.458µs, max_per_file_from_linter: 62.818µs, max_same_issues: 43.95µs, path_shortener: 40.048µs, uniq_by_line: 17.738µs, autogenerated_exclude: 8.707µs, max_from_linter: 6.96µs, skip_files: 1.87µs, diff: 1.622µs 
INFO Memory: 36 samples, avg is 68.3MB, max is 230.0MB 
INFO Execution took 3.869388262s 
```

**Comment by jirfag** [comment]
hi! I guess the problem is in the go version.
Golangci-lint versions > 1.10.2 supports Go 1.10 and 1.11. Golangci-lint versions <= v1.10.2 supported Go 1.9, 1.10, 1.11.

You use new golangci-lint, which doesn't support go 1.9

**Comment by ivila** [comment]
> hi! I guess the problem is in the go version.
> Golangci-lint versions > 1.10.2 supports Go 1.10 and 1.11. Golangci-lint versions <= v1.10.2 supported Go 1.9, 1.10, 1.11.
> 
> You use new golangci-lint, which doesn't support go 1.9

Thanks, it works.
